src/core/core.h

```diff
@@ -126,70 +126,69 @@ public slots:
-
-    void friendLastSeenChanged(uint32_t friendId, const QDateTime& dateTime);
-
-    void emptyGroupCreated(int groupnumber);
-    void groupInviteReceived(uint32_t friendId, uint8_t type, QByteArray publicKey);
-    void groupMessageReceived(int groupnumber, int peernumber, const QString& message, bool isAction);
-    void groupNamelistChanged(int groupnumber, int peernumber, uint8_t change);
-    void groupTitleChanged(int groupnumber, const QString& author, const QString& title);
-    void groupPeerAudioPlaying(int groupnumber, int peernumber);
-
-    void usernameSet(const QString& username);
-    void statusMessageSet(const QString& message);
-    void statusSet(Status status);
-    void idSet(const QString& id);
-    void selfAvatarChanged(const QPixmap& pic);
-
-    void messageSentResult(uint32_t friendId, const QString& message, int messageId);
-    void groupSentResult(int groupId, const QString& message, int result);
-    void actionSentResult(uint32_t friendId, const QString& action, int success);
-
-    void receiptRecieved(int friedId, int receipt);
-
-    void failedToAddFriend(const QString& userId, const QString& errorInfo = QString());
-    void failedToRemoveFriend(uint32_t friendId);
-    void failedToSetUsername(const QString& username);
-    void failedToSetStatusMessage(const QString& message);
-    void failedToSetStatus(Status status);
-    void failedToSetTyping(bool typing);
-
-    void failedToStart();
-    void badProxy();
-
-    void fileSendStarted(ToxFile file);
-    void fileReceiveRequested(ToxFile file);
-    void fileTransferAccepted(ToxFile file);
-    void fileTransferCancelled(ToxFile file);
-    void fileTransferFinished(ToxFile file);
-    void fileUploadFinished(const QString& path);
-    void fileDownloadFinished(const QString& path);
-    void fileTransferPaused(ToxFile file);
-    void fileTransferInfo(ToxFile file);
-    void fileTransferRemotePausedUnpaused(ToxFile file, bool paused);
-    void fileTransferBrokenUnbroken(ToxFile file, bool broken);
-
-    void fileSendFailed(uint32_t friendId, const QString& fname);
+    void connected() const;
+    void disconnected() const;
+
+    void friendRequestReceived(const QString& userId, const QString& message) const;
+    void friendMessageReceived(uint32_t friendId, const QString& message, bool isAction) const;
+
+    void friendAdded(uint32_t friendId, const QString& userId) const;
+    void friendshipChanged(uint32_t friendId) const;
+
+    void friendStatusChanged(uint32_t friendId, Status status) const;
+    void friendStatusMessageChanged(uint32_t friendId, const QString& message) const;
+    void friendUsernameChanged(uint32_t friendId, const QString& username) const;
+    void friendTypingChanged(uint32_t friendId, bool isTyping) const;
+    void friendAvatarChanged(uint32_t friendId, const QPixmap& pic) const;
+    void friendAvatarRemoved(uint32_t friendId) const;
+
+    void friendRemoved(uint32_t friendId) const;
```

Comment:
If a signal is const, it means that it's allowed to call it from a const function.
Signals like friendRemoved shouldn't be const! If a friend was removed, then we *definitely* modified the state.